filters/auth/oidc.go

```diff
@@ -38,8 +40,31 @@ const (
 	stateValidity       = 1 * time.Minute
 	oidcInfoHeader      = "Skipper-Oidc-Info"
 	cookieMaxSize       = 4093 // common cookie size limit http://browsercookielimits.squawky.net/
+
+	// Deprecated: The host of the Azure Active Directory (AAD) graph API
+	azureADGraphHost = "graph.windows.net"
```

Comment:
not sure if we need specific Azure/Microsoft related things.
What does it provide?
If it's required (looks like, but want to be sure), then please refactor code to have these separated in functions.

Reply:
That's unfortunate that we need some customization for Azure. 
1. The endpoint returned by Azure should not be used according to docs https://docs.microsoft.com/en-us/graph/migrate-azure-ad-graph-overview and seeing that in other implementations 
https://github.com/grafana/grafana/blob/main/pkg/login/social/azuread_oauth.go#L183
https://github.com/hashicorp/vault-plugin-auth-jwt/blob/b724ae7ef01864b4f931814d75b875252076936e/provider_azure.go#L112-L120
2. The legacy endpoint does not provide useful information by just returning the GUID instead of group names, hence we have to use another query.

That's certainly the unpleasant part of this PR, happy for ideas how to account for such customizations.

Reply:
Then let's have it, but create a func that patches the settings it needs so we do no pollute too much the main func.

Reply:
After some research we found out that not many services use this oidc function. 
For now we can implement a azure specific one and can let it open for the future to write generic code for more providers because the return json/struct is not strict specified in spec.

Now: azure specific func with no main pollution :-)
Future: use a generic provider with a interface

Is that ok for you? @szuecs and @universam1

Reply:
Thanks for the research @ripx80

Reply:
I don't really get it. :) 
I think types and const are fine like this.

One issue is `snet.NewClient`, which I pointed out should be only instantiated once for every url target host, see https://github.com/zalando/skipper/pull/1967 and https://github.com/zalando/skipper/pull/1962/files#r810945168

and then there is the issue ` if uri.Host == azureADGraphHost { ... }`, which can be just refactored to a func instead of the if body.
I think `resolveDistributedClaim()` could be renamed to  `resolveDistributedClaimAzure()` and I would say good enough, if you add a test case, which might need some work to create the distributed claim targets.

Reply:
Thanks @szuecs  for the detailed specs, that's certainly helpful and sounds reasonable.

Our question was, can we implement a generic, Azure agnostic implementation as well... It turns out that neither from the specs nor from other known IdPs we can discover a concrete implementation of distributed-claims response, therefore only Azure support is what we can offer for now. Maybe leaving for the future extension to support other provider that might become of interest?

Reply:
sgtm!

Reply:
implemented